client/terminal.c

```diff
@@ -44,26 +47,38 @@ int terminal(void)
 		printf("Can't initialize sockets");
 	}
 
-	char cmd[XDAG_COMMAND_MAX];
+    	char cmd[XDAG_COMMAND_MAX];
 	char cmd2[XDAG_COMMAND_MAX];
 
+#if !defined(_WIN32) && !defined(_WIN64)
+	rl_readline_name = "xdag";
+	rl_attempted_completion_function = xdag_com_completion;
+#endif
+    
 	while (1) {
 		int ispwd = 0, c = 0;
-		printf("%s> ", g_progname); fflush(stdout);
-		fgets(cmd, XDAG_COMMAND_MAX, stdin);
+		#if !defined(_WIN32) && !defined(_WIN64)
+        	char *pcmd;
+        	pcmd = readline("xdag> ");
+        	strcpy(cmd, pcmd);
+		#else
+        	printf("%s> ", g_progname);
+        	fflush(stdout);
+        	fgets(cmd, XDAG_COMMAND_MAX, stdin);
+		#endif
 		strcpy(cmd2, cmd);
-		char *ptr = strtok_r(cmd2, " \t\r\n", &lasts);
-		if (!ptr) continue;
-		if (!strcmp(ptr, "exit")) break;
-		if (!strcmp(ptr, "xfer")) {
-			uint32_t pwd[4];
-			xdag_user_crypt_action(pwd, 0, 4, 4);
-			sprintf(cmd2, "pwd=%08x%08x%08x%08x ", pwd[0], pwd[1], pwd[2], pwd[3]);
-			ispwd = 1;
+        	char *ptr = strtok_r(cmd2, " \t\r\n", &lasts);
+        	if (!ptr) continue;
+        	if (!strcmp(ptr, "exit")) break;
+        	if (!strcmp(ptr, "xfer")) {
+            	uint32_t pwd[4];
```

Comment:
Still has spaces and tabs issue. Could you just use tabs instead of spaces?

Reply:
ok